src/Features/CSharp/Portable/AddBraces/CSharpAddBracesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.AddBraces;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.AddBraces
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpAddBracesDiagnosticAnalyzer : AbstractAddBracesDiagnosticAnalyzer<SyntaxKind>
+    {
+        protected override ImmutableArray<SyntaxKind> SyntaxKindsOfInterest { get; } =
+            ImmutableArray.Create(SyntaxKind.IfStatement,
+                SyntaxKind.ElseClause,
+                SyntaxKind.ForStatement,
+                SyntaxKind.ForEachStatement,
+                SyntaxKind.WhileStatement,
+                SyntaxKind.DoStatement,
+                SyntaxKind.UsingStatement);
+
+        public override void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            if (context.Node.IsKind(SyntaxKind.IfStatement))
+            {
+                var ifStatement = (IfStatementSyntax) context.Node;
+                if (AnalyzeIfStatement(ifStatement))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(SupportedDiagnostics[0],
+                        ifStatement.IfKeyword.GetLocation(), "if"));
+                }
+            }
+
+            if (context.Node.IsKind(SyntaxKind.ElseClause))
+            {
+                var elseClause = (ElseClauseSyntax)context.Node;
+                if (AnalyzeElseClause(elseClause))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(SupportedDiagnostics[0],
+                        elseClause.ElseKeyword.GetLocation(), "else"));
```

Comment:
You can use SyntaxFacts.GetText(SyntaxKind.ElseKeyword).  I'd prefer that to hardcoding hte strings.